src/CMakeLists.txt

```diff
@@ -425,41 +425,32 @@ add_library(common-objs OBJECT ${libcommon_files})
 add_dependencies(common-objs legacy-option-headers)
 
 if(WITH_JAEGER)
-  find_package(yaml-cpp 0.6.0)
-  if(NOT yaml-cpp_FOUND)
-    include(Buildyaml-cpp)
-    build_yamlcpp()
-    set(jaeger_libs ${CMAKE_BINARY_DIR}/external/lib/libyaml-cpp.so
-		    ${CMAKE_BINARY_DIR}/external/lib/libyaml-cpp.so.0.6
-		    ${CMAKE_BINARY_DIR}/external/lib/libyaml-cpp.so.0.6.2)
-    execute_process(COMMAND bash -c "grep -q 'yaml-cpp' debian/libjaeger.install || echo 'usr/lib/libyaml-cpp.so.*' >> debian/libjaeger.install"
-		    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
-    add_library(yaml-cpp::yaml-cpp SHARED IMPORTED)
-    add_dependencies(yaml-cpp::yaml-cpp yaml-cpp)
-    set_library_properties_for_external_project(yaml-cpp::yaml-cpp yaml-cpp)
-    list(APPEND dependencies "yaml-cpp")
-  endif()
-  include(IncludeJaeger)
-  add_library(jaeger-base INTERFACE)
-  add_dependencies(common-objs
-  yaml-cpp::yaml-cpp
+  include(BuildJaeger)
+  find_package(thrift 0.13.0)
+  find_package(yaml-cpp 0.6.0 REQUIRED)
+  # will also append thrift if not found
+  list(APPEND jaeger_base
   opentracing::libopentracing
-  thrift::libthrift
   jaegertracing::libjaegertracing)
+
+  build_jaeger()
+
+  include_directories(${CMAKE_BINARY_DIR}/external/include)
+  # for testing jaeger build
+  add_custom_target(jaeger_base DEPENDS ${jaeger_base})
+  add_library(jaeger-base INTERFACE)
+  add_dependencies(common-objs ${jaeger_base})
   target_link_libraries(jaeger-base INTERFACE
-  yaml-cpp::yaml-cpp
-  opentracing::libopentracing
-  thrift::libthrift
-  jaegertracing::libjaegertracing)
-  include_directories(SYSTEM ${CMAKE_BINARY_DIR}/external/include)
-  #with CMake 3.12+ the following can be replaced by:
-  #target_link_libraries(common-objs jaeger-base)
-   list(APPEND jaeger_libs
+    yaml-cpp::yaml-cpp
+    opentracing::libopentracing
+    thrift::libthrift
+    jaegertracing::libjaegertracing)
+
+  list(APPEND jaeger_libs
     ${CMAKE_BINARY_DIR}/external/lib/libjaegertracing.so.0
     ${CMAKE_BINARY_DIR}/external/lib/libjaegertracing.so.0.6.1
     ${CMAKE_BINARY_DIR}/external/lib/libopentracing.so.1
-    ${CMAKE_BINARY_DIR}/external/lib/libopentracing.so.1.6.0
-    ${CMAKE_BINARY_DIR}/external/lib/libthrift.so.0.13.0)
+    ${CMAKE_BINARY_DIR}/external/lib/libopentracing.so.1.6.0)
```

Comment:
nit, i don't think we need to install both 
`${CMAKE_BINARY_DIR}/external/lib/libopentracing.so.1` and `${CMAKE_BINARY_DIR}/external/lib/libopentracing.so.1.6.0`, the latter is enough.